conan/api/subapi/new.py

```diff
@@ -85,10 +86,12 @@ def _read_files(self, template_folder):
     @staticmethod
     def render(template_files, definitions):
         result = {}
+        name = definitions.get("name", "Pkg")
         definitions["conan_version"] = __version__
+        definitions["package_name"] = name.replace("-", "_").replace("+", "_").replace(".", "_")
         for k, v in template_files.items():
-            k = Template(k, keep_trailing_newline=True).render(**definitions)
-            v = Template(v, keep_trailing_newline=True).render(**definitions)
+            k = Template(k, keep_trailing_newline=True, undefined=StrictUndefined).render(**definitions)
```

Comment:
I had to add `StrictUndefined` so that it raises an exception if a context variable is undefined.
That means that you have always to set all the definitions unless the variable has a default filter, that works as it used to work.